lib/lexicon.py

```diff
@@ -49,11 +50,13 @@ def to_xml(self):
         for p in self.phon:
             el = ET.SubElement(res, "phon")
             el.text = p
-        for s in self.synt:
+        if self.synt and any(token.strip() for token in self.synt):
```

Comment:
I would not modify the token sequence here. The lexicon is a data-structure that is written to disk. Having these sort of modifications done here makes it harder to reason about what's going on.

Reply:
Im not sure whether I understand the question. Now synt should be a single sequence of tokens, thus a list of strings. Here what added is only a check to avoid empty token written to final lexicon file. Is empty token in synt fine with RASR?

Reply:
What you say is true. But I think curufinwes argument is, that the check for empty tokens should not happen here in this method, but at the point where the `Lemma` object is created.
Here we would assume that `self.synt`, if given, is correctly formed.

Reply:
Yes, @michelwi is right. The main point is that now for a given lexicon `lex` we do not have `lex == load(store(lex))` as the act of storing changes the data-structure. This is not good behavior in general as you should be able to assume that storing and loading a datastructure should give you the same object.

Reply:
@michelwi @curufinwe Got it, thanks!